plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -547,7 +553,13 @@ public synchronized void replaceTable(String databaseName, String tableName, Tab
 
     public synchronized void renameTable(String databaseName, String tableName, String newDatabaseName, String newTableName)
     {
-        setExclusive((delegate, hdfsEnvironment) -> delegate.renameTable(databaseName, tableName, newDatabaseName, newTableName));
+        setExclusive((delegate, hdfsEnvironment) -> {
+            Table oldTable = getExistingTable(databaseName, tableName);
+            delegate.renameTable(databaseName, tableName, newDatabaseName, newTableName);
+            if (isLocationPresent(oldTable.getStorage())) {
+                directoryCallback.onTableAltered(oldTable, new Path(oldTable.getStorage().getLocation()));
+            }
```

Comment:
What's the case when location is absent?

Reply:
Views